pandas/core/generic.py

```diff
@@ -699,18 +699,100 @@ def pop(self, item):
 
     def squeeze(self, axis=None):
         """
-        Squeeze length 1 dimensions.
+        Squeeze 1 dimensional axis objects into scalars.
+
+        If the given axis consists of one dimensional objects, they are turned
+        into scalars. In case no axis is specified, all axes are subject to
+        squeezing. In any case, objects in axes that can't be squeezed are
+        left unchanged.
 
         Parameters
         ----------
-        axis : None, integer or string axis name, optional
-            The axis to squeeze if 1-sized.
+        axis : integer or string, optional
+            A specific axis to squeeze.
 
             .. versionadded:: 0.20.0
 
         Returns
         -------
-        scalar if 1-sized, else original object
+        DataFrame, Series or scalar
+            The projection after squeezing axis.
+
+        See Also
+        --------
+        Series.iloc : Integer-location based indexing for selecting scalars
+        DataFrame.iloc : Integer-location based indexing for selecting Series
+
+        Examples
+        --------
+        >>> primes = pd.Series([2, 3, 5, 7])
+
+        Slicing might produce a Series with a single value:
+
+        >>> even_primes = primes[primes % 2 == 0]
+        >>> even_primes
+        0    2
+        dtype: int64
+        >>> even_primes.squeeze()
+        2
+
+        Squeezing objects with more than 1 dimension does nothing:
+
+        >>> odd_primes = primes[primes % 2 == 1]
+        >>> odd_primes
+        1    3
+        2    5
+        3    7
+        dtype: int64
+        >>> odd_primes.squeeze()
+        1    3
+        2    5
+        3    7
+        dtype: int64
+
+        Squeezing is even more effective when used with DataFrames.
+
+        >>> df = pd.DataFrame([[1,2], [3, 4]], columns=['a', 'b'])
+        >>> df
+           a  b
+        0  1  2
+        1  3  4
+
+        Slicing a single column will produce a DataFrame with one of the
+        axis having only 1 dimension:
+
+        >>> df_a = df[['a']]
+        >>> df_a
+           a
+        0  1
+        1  3
+
+        Objects along the column are 1 dimensional, so they can be squeezed
+        into scalars:
+
+        >>> df_a.squeeze('columns')
```

Comment:
I think the first example is great but this one could use a little work. With how it's presented it wouldn't make sense to use `squeeze` instead of just selecting the desired `Series`. Perhaps using a predicate here of say `'a' == 1` and squeezing the result of that would showcase the utility better?

Reply:
Makes sense, in most cases squeeze will be necessary because of indirect slicing rather than specific axis selections.